aten/src/ATen/native/cuda/DepthwiseConv3d.cu

```diff
@@ -0,0 +1,634 @@
+    int kKnownDilationT, int kKnownDilationH, int kKnownDilationW>
+__global__ void conv_depthwise3d_cuda_kernel(
+    const PackedTensorAccessor32<scalar_t, 5> input,
+    PackedTensorAccessor32<scalar_t, 5> output,
+    const PackedTensorAccessor32<scalar_t, 5> kernel,
+    const scalar_t* bias,
+    int dT, int dH, int dW,
+    int pT, int pH, int pW,
+    int dilationT_, int dilationH_, int dilationW_)
+{
+  const int kT = kKnownKernelT > 0 ? kKnownKernelT : kernel.size(2);
+  const int kH = kKnownKernelH > 0 ? kKnownKernelH : kernel.size(3);
+  const int kW = kKnownKernelW > 0 ? kKnownKernelW : kernel.size(4);
+  const int oC = output.size(1);
+  const int oT = output.size(2);
+  const int oH = output.size(3);
+  const int oW = output.size(4);
+  const int iC = input.size(1);
+  const int iT = input.size(2);
+  const int iH = input.size(3);
+  const int iW = input.size(4);
+  const int channel_multiplier = oC / iC;
+  const int dilationT = kKnownDilationT > 0 ? kKnownDilationT : dilationT_;
+  const int dilationH = kKnownDilationH > 0 ? kKnownDilationH : dilationH_;
+  const int dilationW = kKnownDilationW > 0 ? kKnownDilationW : dilationW_;
+  const int num_output = output.size(0) * output.stride(0);
+
+  CUDA_KERNEL_LOOP(index, num_output) {
+    const int out_col = index % oW;
+    const int out_row = (index / oW) % oH;
+    const int out_frame = (index / oW / oH) % oT;
+    const int out_channel = (index / oW / oH / oT) % oC;
+    const int batch = index / oW / oH / oT / oC;
+
+    const int in_channel = out_channel / channel_multiplier;
+
+    const int in_col_start = out_col * dW - pW;
+    const int in_row_start = out_row * dH - pH;
+    const int in_frame_start = out_frame * dT - pT;
+
+    accscalar_t sum = 0;
+    const scalar_t *kernel_ptr = kernel[out_channel].data();
+    const scalar_t *input_ptr =
+        &input[batch][in_channel][in_frame_start][in_row_start][in_col_start];
+    for (int k_frame = 0; k_frame < kT; ++k_frame) {
+      const int in_frame = in_frame_start + k_frame * dilationT;
+      for (int k_row = 0; k_row < kH; ++k_row) {
+        const int in_row = in_row_start + k_row * dilationH;
+        for (int k_col = 0; k_col < kW; ++k_col) {
+          const scalar_t op1 = __ldg(kernel_ptr++);
+          const int in_col = in_col_start + k_col * dilationW;
+          if (in_frame >= 0 && in_row >= 0 && in_col >= 0 &&
+              in_frame < iT && in_row < iH && in_col < iW) {
+            sum += op1 * __ldg(input_ptr);
+          }
+          input_ptr += dilationW;
+        }
+        input_ptr += iW * dilationH - kW * dilationW;
+      }
+      input_ptr += iW * (iH * dilationT - kH * dilationH);
+    }
+    if (bias != NULL) {
+      sum += bias[out_channel];
+    }
+
+    output[batch][out_channel][out_frame][out_row][out_col] = sum;
+  }
+}
+
+template <typename scalar_t, typename accscalar_t,
+    int kKnownKernelT, int kKnownKernelH, int kKnownKernelW,
```

Comment:
nit: you are not going to support different templated kernel sizes for different dimensions, so might as well have a single template argument here, same for dilation and stride